src/fr-device/index.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  browserProps,
+  configurableCategories,
+  delay,
+  devicePlatforms,
+  displayProps,
+  fontNames,
+  hardwareProps,
+  platformProps,
+} from './defaults';
+import {
+  BaseProfileConfig,
+  Category,
+  CollectParameters,
+  DeviceProfileData,
+  Geolocation,
+  ProfileConfigOptions,
+} from './interfaces';
+import Collector from './collector';
+
+/**
+ * @class FRDevice - Collects user device metadata
+ *
+ * Example:
+ *
+ * ```js
+ * // Instantiate new device object (w/optional config, if needed)
+ * const device = new forgerock.FRDevice(
+ *   // optional configuration
+ * );
+ * // override any instance methods, if needed
+ * // e.g.: device.getDisplayMeta = () => {};
+ *
+ * // Call getProfile with required argument obj of boolean properties
+ * // of location and metadata
+ * const profile = await device.getProfile({
+ *   location: isLocationRequired,
+ *   metadata: isMetadataRequired,
+ * });
+ * ```
+ */
+class FRDevice extends Collector {
+  config: BaseProfileConfig = {
+    fontNames,
+    devicePlatforms,
+    displayProps,
+    browserProps,
+    hardwareProps,
+    platformProps,
+  };
+
+  constructor(config?: ProfileConfigOptions) {
+    super();
+    if (config) {
+      Object.keys(config).forEach((key: string) => {
+        if (!configurableCategories.includes(key)) {
+          throw new Error('Device profile configuration category does not exist.');
+        }
+        this.config[key as Category] = config[key as Category];
+      });
+    }
+  }
+
+  getBrowserMeta(): { [key: string]: string } {
+    if (!navigator) {
+      console.warn('Cannot collect browser metadata. navigator is not defined.');
+      return {};
+    }
+    return this.reduceToObject(this.config.browserProps, navigator);
+  }
+
+  getBrowserPluginsNames(): string {
+    if (!(navigator && navigator.plugins)) {
+      console.warn('Cannot collect browser plugin information. navigator.plugins is not defined.');
+      return '';
+    }
+    return this.reduceToString(Object.keys(navigator.plugins), navigator.plugins);
+  }
+
+  getDeviceName(): string {
+    if (!navigator) {
+      console.warn('Cannot collect device name. navigator is not defined.');
+      return '';
+    }
+    const userAgent = navigator.userAgent;
+    const platform = navigator.platform;
+
+    switch (true) {
```

Comment:
It's just a way of doing a `switch` on the result of a conditional, rather than directly comparing values in the expression. Though, I am wondering why I didn't use the `string.includes()` syntax below.